.github/workflows/updateVersionsAndCreatePR.yml

```diff
@@ -0,0 +1,89 @@
+        - zowe-v1-lts
+        - zowe-v2-lts
+
+jobs:
+  update_versions_and_create_pr:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+
+      - name: Update Package Lists
+        run: |
+          sudo apt-get update -y
+
+      - name: Create Branch
+        run: |
+          version_name="${{ github.event.inputs.version }}"
+          branch_name="update-versions-${version_name}"
+          git checkout -b "$branch_name"
+
+      - name: Get and Change Component Versions
+        id: get_versions
+        run: |
+          components=("perf-timing" "imperative" "cli-test-utils" "secrets-for-zowe-sdk" "core-for-zowe-sdk" "zos-uss-for-zowe-sdk" "provisioning-for-zowe-sdk" "zos-console-for-zowe-sdk" "zos-files-for-zowe-sdk" "zos-logs-for-zowe-sdk" "zosmf-for-zowe-sdk" "zos-workflows-for-zowe-sdk" "zos-jobs-for-zowe-sdk" "zos-tso-for-zowe-sdk" "cli" "cics-for-zowe-cli" "db2-for-zowe-cli" "ims-for-zowe-cli" "mq-for-zowe-cli" "zos-ftp-for-zowe-cli")
+          zowe_version="${{ github.event.inputs.zowe_version }}"
+          version="${{ github.event.inputs.version }}"
+          if [[ "$zowe_version" == "zowe-v1-lts" ]]; then
+            components+=("secure-credential-store-for-zowe-cli")
+          fi
+          # echoing to txt file to later add this file as the PR description
+          echo "This PR updates the following package versions in zowe-versions.yaml for \`$zowe_version\`" >> description.txt
+          echo "| Updated Package | Old Version | New Version |" >> description.txt
+          echo "| --- | --- | --- |" >> description.txt
+          for component in "${components[@]}"; do
+            current_version=$(grep -A 2 "^  $component:" zowe-versions.yaml | grep "$zowe_version" | awk '{print $2}')
+            new_version=$(npm view @zowe/$component "dist-tags.$zowe_version")
+            if [[ "$new_version" != "$current_version" ]]; then
+              echo "|$component | $current_version | $new_version|" >> description.txt
+              # Replaces the version number. If it doesn't match on the first line after the component, it goes to the next line and checks again.
+              sed -i -E "/^  $component:/ { n; s/($zowe_version: ).*/\1$new_version/; t; n; s/($zowe_version: ).*/\1$new_version/ }" zowe-versions.yaml
+            fi
+          done
+
+          # Update zowe-versions.yaml tags section
+          sed -i -E "/$zowe_version:/ { n; s/(version: ).*/\1$version/ }" zowe-versions.yaml
+
+
+      - name: Commit and Push Changes
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "actions@github.com"
+          git commit -am "Update component versions"
+          git push origin HEAD
+
+      - name: Create Pull Request
+        env:
+          GH_TOKEN: ${{ secrets.GHATOKEN }}

```

Comment:
Can we use the default token `secrets.GITHUB_TOKEN`? Or was there a reason that one didn't work - maybe it doesn't have adequate permissions to create pull requests?

Reply:
I just created my own token in testing. these secrets just needed to be switched out pretty sure. i can do that now :)